dev/com.ibm.ws.security.jwtsso_fat/fat/src/com/ibm/ws/security/jwtsso/fat/ConfigAttributeTests.java

```diff
@@ -0,0 +1,483 @@
+     */
+    @Mode(TestMode.LITE)
+    @Test
+    public void test_validConsumerRef() throws Exception {
+        reconfigServer("server_testgoodconsumer.xml");
+        Expectations expectations = new Expectations();
+        expectations.addExpectations(getSuccessfulLoginPageExpectations(ACTION_INVOKE_PROTECTED_RESOURCE));
+
+        String protectedUrl = "http://" + server.getHostname() + ":" + server.getHttpDefaultPort() + "/formlogin/SimpleServlet";
+
+        WebClient wc = new WebClient();
+        Page response = invokeUrl(wc, protectedUrl); // get back the login page
+        validationUtils.validateResult(response, ACTION_INVOKE_PROTECTED_RESOURCE, expectations);
+
+        response = performLogin(response);
+
+        String responseStr = response.getWebResponse().getContentAsString();
+        boolean check = responseStr.contains("\"iss\":\"https://flintstone:19443/jwt/defaultJWT\"");
+        assertTrue("Issuer in token did not match the one configured in the builder", check);
+
+        // now access resource a second time, force token to be examined by consumer
+        response = invokeUrl(wc, protectedUrl);
+        responseStr = response.getWebResponse().getContentAsString();
+        boolean check2 = responseStr.contains("SimpleServlet");
+        assertTrue("Did not access protected resource with custom consumer", check2);
+
+    }
+
+    /**
+     * Test the jwtConsumerRef attribute. Specify an invalid consumer and try to authenticate.
+     * We should get an error message about the invalid consumer.
+     *
+     * @throws Exception
+     */
+    @Test
+    @Mode(TestMode.LITE)
+    @AllowedFFDC({ "com.ibm.ws.security.authentication.AuthenticationException",
+                   "com.ibm.websphere.security.WSSecurityException",
+                   "com.ibm.websphere.security.jwt.InvalidConsumerException" })
+    public void test_invalidConsumerRef() throws Exception {
+        reconfigServer("server_testbadconsumer.xml");
+        Expectations expectations = new Expectations();
+        expectations.addExpectations(getSuccessfulLoginPageExpectations(ACTION_INVOKE_PROTECTED_RESOURCE));
+
+        String protectedUrl = "http://" + server.getHostname() + ":" + server.getHttpDefaultPort() + "/formlogin/SimpleServlet";
+
+        WebClient wc = new WebClient();
+        Page response = invokeUrl(wc, protectedUrl); // get back the login page
+        validationUtils.validateResult(response, ACTION_INVOKE_PROTECTED_RESOURCE, expectations);
+
+        response = performLogin(response); // should fail and we should get login page again
+
+        String responseStr = response.getWebResponse().getContentAsString();
+        boolean check = responseStr.contains("Form Login Page");
+        assertTrue("Did not receive login page on failure", check);
+
+        String errorMsg = server.waitForStringInLogUsingMark("CWWKS6030E", 100);
+        assertFalse("Did not find expected error message CWWKS6030E in log", errorMsg == null);
```

Comment:
Another one where a message constant would be helpful, e.g. `CWWKS6030E_JWT_CONSUMER_CONFIG_NOT_FOUND`.